cpp/tests/interpolate/cubic_spline_test.cpp

```diff
@@ -14,202 +14,105 @@
-        EXPECT_EQ(d_expect[i][j], host_data[j]);
-      }
-    }
+TEST_F(CubicSplineTest, test_coefficients_single) {
+  cudf::test::fixed_width_column_wrapper<float> t_column{{0, 1, 2, 3, 4}};
+  cudf::test::fixed_width_column_wrapper<float> y_column{{3, 2, 3, 4, 3}};
+  cudf::test::fixed_width_column_wrapper<int> ids_column{{0, 0}};
+  cudf::test::fixed_width_column_wrapper<int> prefix_column{{0, 5}};
+
+  auto splines = cuspatial::cubicspline_coefficients(t_column, y_column,
+                                                     ids_column, prefix_column);
+
+  auto d_expect = get_d_expect();
+  for (unsigned int i = 0; i < d_expect.size(); ++i) {
+    cudf::test::expect_columns_equivalent(
+        splines->get_column(i), cudf::test::fixed_width_column_wrapper<float>(
+                                    d_expect[i].begin(), d_expect[i].end()));
+  }
 }
 
-TEST_F(CubicSplineTest, test_coefficients_full)
-{
-    int point_len = 15;
-    float t[point_len] = {0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 0, 1, 2, 3, 4};
-    float y[point_len] = {3, 2, 3, 4, 3, 3, 2, 3, 4, 3, 3, 2, 3, 4, 3};
-    int ids_len = 4;
-    int ids[ids_len] = {0, 0, 1, 2};
-    int prefix[ids_len] = {0, 5, 10, 15};
- 
-    cudf::column t_column = make_device_column<float>(t, point_len);
-    cudf::column y_column = make_device_column<float>(y, point_len);
-    cudf::column ids_column = make_device_column<int>(ids, ids_len);
-    cudf::column prefix_column = make_device_column<int>(prefix, ids_len);
-
-    std::unique_ptr<cudf::experimental::table> splines =
-        cuspatial::cubicspline_coefficients(
-            t_column,
-            y_column,
-            ids_column,
-            prefix_column
-        );
-
-    auto d_expect = get_d_expect();
-    for(unsigned int i = 0 ; i < d_expect.size() ; ++i){
-      cudf::column_view device_column = splines->view().column(i);
-      std::vector<float> host_data;
-      host_data.resize(device_column.size());
-      cudaMemcpy(host_data.data(), device_column.data<float>(),
-            device_column.size() * sizeof(float),
-            cudaMemcpyDeviceToHost);
-      for(unsigned int j = 0 ; j < host_data.size() ; ++j ){
-        EXPECT_EQ(d_expect[i][j%d_expect[i].size()], host_data[j]);
-      }
+TEST_F(CubicSplineTest, test_coefficients_full) {
+  cudf::test::fixed_width_column_wrapper<float> t_column{
+      {0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 0, 1, 2, 3, 4}};
+  cudf::test::fixed_width_column_wrapper<float> y_column{
+      {3, 2, 3, 4, 3, 3, 2, 3, 4, 3, 3, 2, 3, 4, 3}};
+  cudf::test::fixed_width_column_wrapper<int> ids_column{{0, 0, 1, 2}};
+  cudf::test::fixed_width_column_wrapper<int> prefix_column{{0, 5, 10, 15}};
+
+  auto splines = cuspatial::cubicspline_coefficients(t_column, y_column,
+                                                     ids_column, prefix_column);
+
+  auto d_expect = get_d_expect();
+  for (unsigned int i = 0; i < d_expect.size(); ++i) {
+    auto h_expected_col = d_expect[i];
```

Comment:
I think all the h_ and d_ are unnecessary, e.g.

```suggestion
    auto expected_col = this->reference_spline[i];
```

Reply:
This is a really accurate catch - in this case `h_` refers to host, `d_` refers to implementation naming, so I totally agree.